### Hello everyone! I know this is not a Tumblr post, but this seemed a bit more adequate for me!

# Code
Below you can find the code that I used as well as its output:

In [1]:
import numpy as np
import pandas as pd
import statsmodels.formula.api as smf
import statsmodels.stats.multicomp as multi

data=pd.read_csv("nesarc.csv",low_memory=False)

#Data for Total Personal Income in the last 12 months (Ages 20-25) (Income between 0=25000$)
A1 = data[(data['AGE'] >= 20) & (data['AGE'] <= 25) & (data['S1Q10B'] >= 0) & (data['S1Q10B'] <= 7)]
B1 = A1.copy()


recode= {0:0,1:4999,2:7999,3:9999,4:12999,5:14999,6:19999,7:24999}
B1['INCOMERANGE']=B1['S1Q10B'].map(recode)
d1=B1['INCOMERANGE'].value_counts(sort=False)

#CONVERT TO NUMERIC
B1['INCOMERANGE']=B1['INCOMERANGE'].apply(pd.to_numeric, errors='coerce')

# using ols function for calculating the F-statistic and associated p value
model1 = smf.ols(formula='INCOMERANGE ~ C(DMAJORDEPSNI12)', data=B1)
results1 = model1.fit()
print (results1.summary())

B2 = B1[['INCOMERANGE', 'DMAJORDEPSNI12']].dropna()

print ('Means for INCOMERANGE by Major Depression status')
m1= B2.groupby('DMAJORDEPSNI12').mean()
print (m1)

print ('Standard Deviations for INCOMERANGE by Major Depression status')
sd1 = B2.groupby('DMAJORDEPSNI12').std()
print (sd1)

                            OLS Regression Results                            
Dep. Variable:            INCOMERANGE   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                 -0.000
Method:                 Least Squares   F-statistic:                    0.3300
Date:                Fri, 10 Nov 2023   Prob (F-statistic):              0.566
Time:                        19:54:15   Log-Likelihood:                -36390.
No. Observations:                3510   AIC:                         7.278e+04
Df Residuals:                    3508   BIC:                         7.280e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                             coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------
Intercept               1.19

# Examining the results

When examining the association between low income young adults (quantitative response) and major depression in the last 12 months between young adults (categorical explanatory), an Analysis of Variance (ANOVA) revealed that among low income young adults (my sample), those young adults who said Yes, they are experiencing major depression reported lesser income (Mean=11730.31, s.d. ±7593.49) compared to those people who don’t experience major depression (Mean=11961.97, s.d. ±7712.05).

My F-statistic is 0.33 and my P-value is 0.566 which is much higher than our alpha therefore, meaning that **we cannot reject** the null Hypothesis (Ho), meaning that there is **no association** between low income young adults and major depression.

Thank you very much for your time!!